### PR TITLE
docs: refinements

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
Adds some minor refactors and simplifications (i.e. using wait strategies instead of sleep)

## Why is it important?

Add progress on the original PR without too many ping-pong in the GH UI 😅

### CODE REVIEW

File: docker_files_test.go [hunk 1]
```diff
@@ -83,17 +81,9 @@ func TestCopyFileToRunningContainer(t *testing.T) {
...
 
 	// Give some time to the wait script to catch the hello script being created
-	time.Sleep(200 * time.Millisecond)
-
-	logs, err := container.Logs(ctx)
-	require.NoError(t, err)
-
-	logsString := new(strings.Builder)
-	_, err = io.Copy(logsString, logs)
+	err = wait.ForLog("done").WithStartupTimeout(200*time.Millisecond).WaitUntilReady(ctx, container)
```
Comment (Minivera) [inline docker_files_test.go:84]: 😮 I didn't know you could use wait strategies like this!

File: docker_files_test.go [hunk 2]
```diff
@@ -161,9 +150,12 @@ func TestCopyDirectoryToRunningContainerAsFile(t *testing.T) {
 		},
 		Started: true,
 	})
+	require.NoError(t, err)
```
Comment (Minivera) [inline docker_files_test.go:153]: The examples I saw on the testcontainer website didn't leak the assertion library, which is why I kind of ignored the errors. Maybe:

```suggestion
if err != nil {
		t.Fatal(err)
	}
```

Would be best here?

**Review by Minivera — APPROVED** [review]
Small comments and questions on some of the decisions. Thank you for the updates 😄

File: docker_files_test.go (hunk 1 above)
Comment (mdelapenya) [reply docker_files_test.go:84]: I think this is cool to use, and more deterministic than a fixed sleep, that could be different/slower/faster at CI time

File: docker_files_test.go (hunk 2 above)
Comment (mdelapenya) [reply docker_files_test.go:153]: Mmm you are right, let me revert that single commit 🙏

Comment (mdelapenya) [reply docker_files_test.go:153]: One question: is it fine to leak the testing library?

**Comment by mdelapenya** [comment]
oh I pushed the branch and that could have merged the branch??? Feel free to revert the last commit into your branch. Sorry

**Pull request merged**